src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java

```diff
@@ -75,26 +77,50 @@ public SalesforceBatchMultiSource(SalesforceMultiSourceConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
-    config.validate(stageConfigurer.getFailureCollector()); // validate before macros are substituted
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    if (!config.getConnection().canAttemptToEstablishConnection()) {
+      config.validateFilters(collector);
```

Comment:
Why do we need a separate call to validateFilters() here ?

Reply:
In case of not able to connect with Salesforce, we need to basic validations which doesn't require connection to be established.